docs/DebuggerHTML-Components-Reducers-Actions.md

```diff
@@ -0,0 +1,663 @@
+
+This document gives a detailed view of the components, actions and reducers that make up the debugger.html project. Prior knowledge of React and Redux  is suggested.

React documentation can be found [here](https://facebook.github.io/react/docs/getting-started.html).
Redux documentation can be found [here](http://redux.js.org/).

As with most documentation related to code, this document may be out of date. The last edit date occurred on August 30, 2016.
+
+#Architecture
+
+
+Debugger.html is a React-Redux based application — the UI is constructed using React Components. the follow illustration provides a simplictic high level view:
+
+![](https://docs.google.com/drawings/d/1JTDI-62CG29M37rpTGIDh70rOTuCmJf1VqxCKPe9zxM/pub?w=960&h=720)
+[Click here to Edit](https://docs.google.com/drawings/d/1JTDI-62CG29M37rpTGIDh70rOTuCmJf1VqxCKPe9zxM/edit?usp=sharing)
+
+Application-critical objects are stored in one
+global state object (housed in a Redux store) that some components have
+access to. Many components are not aware of this state but are passed in
+values to render using React properties.
+
+In the Debugger.html project we
+also often use React’s setState to manage component local state. For
+example, storing the state of a tree in the sources view or using it in
+the App component to remember if the file search box is being displayed
+(cmd-&gt;p).
+
+When a user
+manipulates the UI, Redux Actions are fired to collect payload
+data, which affects the state of the application for the given
+operation. Actions set a specific type of operation for the store and
+dispatch the event.
+
+Reducers handle all actions and decide the new
+application state based on the action type. You can think of a reducer
+as a set of functions that take a specific action type and the current
+state of the app as parameters and returns the new state of the
+application.
+
+The Store is a JavaScript object that contains and manages
+the state of the application. After the Reducers create a new version
+of the state, the store will fire a re-rendering of all the
+components. Note that a new state is created every time — the old state
+is not modified.
+
+React uses a Virtual DOM; only required changes to the
+actual DOM will be rendered.
+
+#Component Presentation
+
+
+Debbuger.html uses React Components to render portions of the
+application. Each component’s source code is located under the
+public/js/components folder. In this section we will cover how the
+presentation pieces fit together; later we will discuss
+how debugger.html uses Redux to wire up data to each of the components.
+
+The top-level component is the App component; it encapsulates all
+other components. Presented below is an overview of the component

```

Comment:
I think you can link to it with a relative directory `(../public/js/components)`:

Each component’s source code is located under the [public/js/components](../public/js/components) folder.